streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/TaskSchedulerImpl.scala

```diff
@@ -81,11 +81,12 @@ class TaskSchedulerImpl(appId : Int, config: Config)  extends TaskScheduler {
 
   def setTaskDAG(dag: DAG): Unit = {
     dag.processors.foreach { params =>
-      val (taskGroupId, taskDescription) = params
+      val (processorId, taskDescription) = params
       0.until(taskDescription.parallelism).map((taskIndex: Int) => {
-        val taskId = TaskId(taskGroupId, taskIndex)
+        val taskId = TaskId(processorId, taskIndex)
         val locality = taskLocator.locateTask(taskDescription)
-        val taskLaunchData = TaskLaunchData(taskId, taskDescription, dag.subGraph(taskGroupId))
+        val taskLaunchData = TaskLaunchData(taskId, taskDescription,
+          Subscriber.of(processorId, dag))
```

Comment:
shall we move Subscriber.of(processorId, dag) out of the loop? It seems it is processor specific instead of task specific.